alacritty_terminal/src/term/mod.rs

```diff
@@ -1925,42 +1842,42 @@ impl<T: EventListener> Handler for Term<T> {
     #[inline]
     fn clear_screen(&mut self, mode: ansi::ClearMode) {
         trace!("Clearing screen: {:?}", mode);
-        let template = self.cursor.template;
+        let template = self.grid.cursor.template;
 
         // Remove active selections
-        self.grid.selection = None;
+        self.selection = None;
```

Comment:
Note that for proper testing, it might be useful to disable this reset. At least with shells like `zsh` the selection will otherwise be cleared anyways.

It shouldn't be a problem though, since it will just start working as soon as @nixpulvis has finished work on https://github.com/alacritty/alacritty/pull/3498.